### DISCUSSION TITLE
Trouble getting quicksync using docker-compose to work

### DISCUSSION THREAD

**Discussion by Akashic101** [photoprism/photoprism#4531] [Q&A]
I recently upgraded my system to a i5-11400 and want to see if quicksync is better than ffmpeg 7 (I just upgraded to Ubuntu 24.04). However I have trouble getting the docker-compose.yaml to work properly, following the docs I have this:

```yaml
HOME: "/photoprism" # do not change
PHOTOPRISM_FFMPEG_ENCODER: "intel"
PHOTOPRISM_INIT: "intel"
  devices:
    - "/dev/dri:/dev/dri"
  working_dir: "/photoprism" # do not change
```

However running this using `docker-compose up -d` results in the error `Error response from daemon: error gathering device information while adding custom device "/dev/dri": no such file or directory`

The output of `ls -l /dev/dri` is

```
insgesamt 0
drwxr-xr-x  2 root root         80 Sep 16 11:15 by-path
crw-rw----+ 1 root render 226,   0 Sep 16 11:15 card0
crw-rw----+ 1 root render 226, 128 Sep 16 11:15 renderD128
```

I also checked if the drivers are installed properly, `vainfo` outputs

```
libva info: VA-API version 1.20.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_20
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.20 (libva 2.12.0)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 24.1.0 ()
vainfo: Supported profile and entrypoints
      VAProfileNone                   : VAEntrypointVideoProc
      VAProfileNone                   : VAEntrypointStats
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Simple            : VAEntrypointEncSlice
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointEncSlice
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSlice
      VAProfileH264Main               : VAEntrypointFEI
      VAProfileH264Main               : VAEntrypointEncSliceLP
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSlice
      VAProfileH264High               : VAEntrypointFEI
      VAProfileH264High               : VAEntrypointEncSliceLP
      VAProfileVC1Simple              : VAEntrypointVLD
      VAProfileVC1Main                : VAEntrypointVLD
      VAProfileVC1Advanced            : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointEncPicture
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline: VAEntrypointFEI
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSliceLP
      VAProfileHEVCMain               : VAEntrypointVLD
      VAProfileHEVCMain               : VAEntrypointEncSlice
      VAProfileHEVCMain               : VAEntrypointFEI
      VAProfileHEVCMain               : VAEntrypointEncSliceLP
      VAProfileHEVCMain10             : VAEntrypointVLD
      VAProfileHEVCMain10             : VAEntrypointEncSlice
      VAProfileHEVCMain10             : VAEntrypointEncSliceLP
      VAProfileVP9Profile0            : VAEntrypointVLD
      VAProfileVP9Profile1            : VAEntrypointVLD
      VAProfileVP9Profile2            : VAEntrypointVLD
      VAProfileVP9Profile3            : VAEntrypointVLD
      VAProfileHEVCMain12             : VAEntrypointVLD
      VAProfileHEVCMain12             : VAEntrypointEncSlice
      VAProfileHEVCMain422_10         : VAEntrypointVLD
      VAProfileHEVCMain422_10         : VAEntrypointEncSlice
      VAProfileHEVCMain422_12         : VAEntrypointVLD
      VAProfileHEVCMain422_12         : VAEntrypointEncSlice
      VAProfileHEVCMain444            : VAEntrypointVLD
      VAProfileHEVCMain444            : VAEntrypointEncSliceLP
      VAProfileHEVCMain444_10         : VAEntrypointVLD
      VAProfileHEVCMain444_10         : VAEntrypointEncSliceLP
      VAProfileHEVCMain444_12         : VAEntrypointVLD
      VAProfileHEVCSccMain            : VAEntrypointVLD
      VAProfileHEVCSccMain            : VAEntrypointEncSliceLP
      VAProfileHEVCSccMain10          : VAEntrypointVLD
      VAProfileHEVCSccMain10          : VAEntrypointEncSliceLP
      VAProfileHEVCSccMain444         : VAEntrypointVLD
      VAProfileHEVCSccMain444         : VAEntrypointEncSliceLP
      VAProfileAV1Profile0            : VAEntrypointVLD
      VAProfileHEVCSccMain444_10      : VAEntrypointVLD
      VAProfileHEVCSccMain444_10      : VAEntrypointEncSliceLP
```

The kernel modules are also properly loaded using `lsmod | grep i915` outputs

```
drm_buddy              20480  2 xe,i915
ttm                   110592  3 drm_ttm_helper,xe,i915
drm_display_helper    237568  2 xe,i915
cec                    94208  3 drm_display_helper,xe,i915
i2c_algo_bit           16384  2 xe,i915
video                  73728  2 xe,i915
```

Any idea what I need to do to get this working?

**Comment by Akashic101** [comment]
From [another post on Reddit ](https://old.reddit.com/r/photoprism/comments/1fi991n/trouble_getting_quicksync_using_dockercompose_to/) I found out that `PHOTOPRISM_INIT` doesn't support `intel` as an argument, only `PHOTOPRISM_INIT: "http gpu tensorflow" # Options: "update https gpu tensorflow davfs clitools clean"`. Is this a case where the docs or the docker-compose wasn't updated? Still looking for a solution to use quicksync

**Reply by lastzero** [reply]
https://github.com/photoprism/photoprism/blob/develop/scripts/dist/Makefile

**Reply by Akashic101** [reply]
Some more info regarding this would be appreciated. I see there is a command called `install-intel` in there but I have never worked with a Makefile before. Do I need this to build a custom Docker-image or something? And what does my docker-compose need to look like after that?

**Reply by Akashic101** [reply]
I did some research and used `make intel` to install the necessary drivers and such and changed my `docker-compose.yml` back to the one mentioned in my original comment. However I still get following error:

`Error response from daemon: error gathering device information while adding custom device "/dev/dri": no such file or directory` 

Any idea why docker is unable to find the device, even if everything should work correctly?

**Reply by lastzero** [reply]
If you can't mount /dev/dri then try mounting /dev/dri/renderD128 as shown in the device listing you posted above.

**Reply by Akashic101** [reply]
Sadly this didn't seem to help, I also tried to use `card0` just to rule out all options but that also didn't work. Just for giggles I tried `sudo docker-compose up -d` which did launch but all of my media was gone, I assume cause `sudo` makes it the container run somewhere else. Could this be a permission-issue then? I also tried `chmod 666 /dev/dri/renderD128` as mentioned in [this post](https://unix.stackexchange.com/questions/747523/docker-permissions-issue-accessing-dev-dri-device) but that didn't seem to help as well

**Reply by Akashic101** [reply]
Small update: I tried following the instructions in [this thread](https://forum.jellyfin.org/t-solved-intel-quicksync-on-docker), getting the group id of render prints out `render:x:110:plex,davidmoll`, I then tried adding the group to Photoprism with these changes in the `docker-compose`

```yaml
devices:
  - "/dev/dri/renderD128:/dev/dri/renderD128"
group_add:
  - "110"
```

but to no avail. Some additional info, `id davidmoll` prints out

`uid=1000(davidmoll) gid=1000(davidmoll) Gruppen=1000(davidmoll),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev),110(render),114(ssl-cert),122(lpadmin),135(lxd),136(sambashare)` 

while `id plex` prints out

`uid=999(plex) gid=998(plex) Gruppen=998(plex),44(video),46(plugdev),110(render),1000(davidmoll)`

They both seem to be in the render group. Does Photoprism run under a different user? I don't see any docker-user with `getent passwd`
